configure

```diff
@@ -0,0 +1,338 @@
+EPREFIX="/"
+CXXOPTS=""
+LDFLAGS=""
+SANITIZEOPTS=""
+SBINDIR="/sbin"
+MANDIR="/share/man/"
+SYSCONTROLSOCKET="/run/dinitctl"
+NO_SANITIZE="auto"
+SHUTDOWN_PREFIX=""
+if [ "$PLATFORM" = "Linux" ]; then
+    BUILD_SHUTDOWN=yes
+    SUPPORT_CGROUPS=1
+else
+    BUILD_SHUTDOWN=no
+    SUPPORT_CGROUPS=0
+fi
+
+## Flag praser
+for arg in "$@"; do
+    case "$arg" in
+        --help) usage ;;
+        --quiet) info() { true; } && sub_info() { true; } && warning() { true; } && sub_warning() { true; } ;;
+        --clean) rm -f test* & rm -f mconfig && exit 0 ;;
+        --platform=*) PLATFORM="${arg#*=}" && shift ;;
+        --prefix=*) PREFIX="${arg#*=}" && shift ;;
+        --exec-prefix=*) EPREFIX="${arg#*=}" && shift;;
+        --sbindir=*) SBINDIR="${arg#*=}" && shift ;;
+        --mandir=*) MANDIR="${arg#*=}" && shift ;;
+        --syscontrolsocket=*) SYSCONTROLSOCKET="${arg#*=}" && shift ;;
+        --shutdown-prefix=*) SHUTDOWN_PREFIX="${arg#*=}" && shift ;;
+        --enable-shutdown) BUILD_SHUTDOWN=yes ;;
+        --disable-shutdown) BUILD_SHUTDOWN=no ;;
+        --enable-cgroups) SUPPORT_CGROUPS=1 ;;
+        --disable-cgroups) SUPPORT_CGROUPS=0 ;;
+        --enable-sanitize) NO_SANITIZE=0 ;;
+        --disable-sanitize) NO_SANITIZE=1 ;;
+        --enable-utmpx) USE_UTMPX=1 ;;
+        --disable-utmpx) USE_UTMPX=0 ;;
+        --disable-cxxopts) NO_CXXOPTS=true ;;
+        --disable-ldflags) NO_LDFLAGS=true ;;
+        -*) warning Unknown argument: "$arg" ;;
+        *=*) eval "$arg" ;;
+        *) continue ;;
+    esac
+done
+
+## Resolve final paths
+SBINDIR="$EPREFIX/$SBINDIR"
+MANDIR="$PREFIX/$MANDIR"
+
+## Verify PLATFORM value
+if [ "$PLATFORM" != "Linux" ] && [ "$PLATFORM" != "FreeBSD" ] && \
+[ "$PLATFORM" != "OpenBSD" ] && [ "$PLATFORM" != "Darwin" ]; then
+    warning "$PLATFORM" platform is unknown!
+    sub_warning Known Platforms are: Linux, FreeBSD, OpenBSD, Darwin
+fi
+
+## Create testfile.cc to test c++ compiler
+create_testfile_cc
+
+## Find and test C++ compiler
+info Checking C++ compiler...
+if [ -n "${CXX:-}" ]; then
+    sub_info "$CXX" is user/system prefered C++ compiler.
+    compiler="$CXX"
+else
+    for guess in g++ clang++ c++; do
+        findcxx "$guess"
+        if [ -n "${compiler:-}" ]; then
+            sub_info "$compiler"
+            break # Found
+        fi
+    done
+    if [ -z "${compiler:-}" ]; then
+       sub_error No C++ compiler found! # Not found
+    fi
+fi
+if [ -n "${HOSTCXX:-}" ]; then
+    info Checking Host C++ Compiler...
+    sub_info "$HOSTCXX" is user/system prefered native C++ compiler.
+    hostcompiler="$HOSTCXX"
+    if [ -z "${compiler:-}" ] || [ -z "${hostcompiler:-}" ]; then
+       sub_error No C++ compiler found! # Not found
+    fi
+fi
+cxx_works "$compiler"
+if [ -n "${hostcompiler:-}" ]; then
+    cxx_works "$hostcompiler"
+fi
+
+## Find depends
+if [ "$PLATFORM" = "Linux" ] || [ "$PLATFORM" = "Darwin" ]; then
+    info Checking make is executable...
+    find_program make
```

Comment:
why are you trying to find 'make' (or 'gmake')?

Reply:
I.e. make must be there to build. It doesn't matter what it is called; the user will have to invoke it anyway. I don't think you should fail configure just because you can't find the make executable. The user should know where it is.